src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -242,6 +242,7 @@ public override INamedTypeSymbol CreateErrorTypeSymbol(INamespaceOrTypeSymbol co
             bool isSubmission)
         {
             Debug.Assert(options != null);
+            Debug.Assert(!isSubmission || options.ReferencesSupersedeLowerVersions);
```

Comment:
@cston I'm not sure I understand your question.  I think the only caller that sets `isSubmission == true` also force `ReferencesSupersedeLowerVersions  == true`.  I guess it's possible that someone (internal) could subsequently use `WithOptions` to override that, so perhaps there should also be an assert in `WithOptions`.  I specifically didn't use an exception since the option is internal and so can only be set incorrectly through programmer error.